big_tests/tests/gdpr_SUITE.erl

```diff
@@ -1092,14 +1154,180 @@ retrieve_pubsub_subscriptions(Config) ->
+
+        AffChange = [{Bob, <<"publish-only">>}],
+        pubsub_tools:set_affiliations(Alice, Node1, AffChange, []),
+
+        maybe_stop_and_unload_module(mod_pubsub, mod_pubsub_db_backend, Config),
+        {0, _} = unregister(Bob, Config),
+
+        AliceU = escalus_utils:jid_to_lower(escalus_client:username(Alice)),
+        AliceS = escalus_utils:jid_to_lower(escalus_client:server(Alice)),
+        mongoose_helper:wait_until(
+          fun() ->
+                  mongoose_helper:successful_rpc(mod_pubsub, get_personal_data,
+                                                 [AliceU, AliceS])
+          end,
+          [{pubsub_payloads,["node_name","item_id","payload"],[]},
+           {pubsub_nodes,["node_name","type"],[[NodeName, <<"flat">>]]},
+           {pubsub_subscriptions,["node_name"],[]}]
+         )
+        end).
+
+remove_pubsub_all_data(Config) ->
+    escalus:fresh_story(Config, [{alice, 1}, {bob, 1}], fun(Alice, Bob) ->
+        [Node1={_,Name1}, Node2={_,Name2}, Node3={_,Name3}, Node4={_,Name4}]
+            = pubsub_tools:create_node_names(4),
+        PepNode = make_pep_node_info(Alice, <<"myns">>),
+        pubsub_tools:create_nodes([
+                      {Alice, Node1, []},
+                      {Alice, Node2, []},
+                      {Alice, PepNode, []},
+                      {Bob, Node3, []},
+                      {Bob, Node4, [{type, <<"push">>}]}
+                     ]),
+
+        AffChange = [{Bob, <<"publish-only">>}],
+        pubsub_tools:set_affiliations(Alice, Node1, AffChange, []),
+        pubsub_tools:subscribe(Bob, Node2, []),
+        pubsub_tools:subscribe(Alice, Node3, []),
+
+        {BinItem1, _} = item_content(<<"Item1Data">>),
+        {BinItem2, _} = item_content(<<"Item2Data">>),
+        {BinItem3, _} = item_content(<<"Item3Data">>),
+        {BinItem4, _} = item_content(<<"Item4Data">>),
+        AliceToNode1 = <<"Alice publishes to Node1, but nobody is subscribed">>,
+        AliceToNode2 = <<"Alice published to Node2, so Bob receives it">>,
+        BobToNode1 = <<"Bob publishes to Node1, but nobody is subscribed">>,
+        BobToNode3 = <<"Bob publishes to Node3, so Alice receives it">>,
+
+        pubsub_tools:publish(Alice, AliceToNode1, Node1, [{with_payload, {true, BinItem1}}]),
+
+        pubsub_tools:publish(Alice, AliceToNode2, Node2, [{with_payload, {true, BinItem2}}]),
+        pubsub_tools:receive_item_notification(Bob, AliceToNode2, Node2, []),
+
+        pubsub_tools:publish(Bob, BobToNode1, Node1, [{with_payload, {true, BinItem3}}]),
+
+        pubsub_tools:publish(Bob, BobToNode3, Node3, [{with_payload, {true, BinItem4}}]),
+        pubsub_tools:receive_item_notification(Alice, BobToNode3, Node3, []),
+
+        maybe_stop_and_unload_module(mod_pubsub, mod_pubsub_db_backend, Config),
+        {0, _} = unregister(Alice, Config),
+
+        AliceU = escalus_utils:jid_to_lower(escalus_client:username(Alice)),
```

Comment:
This and the block below are so similar that a helper function would increase readability.